cleanup-github-members/index.js

```diff
@@ -0,0 +1,176 @@
+    console.log(err);
+    fs.appendFileSync(
+      "cleanup-github.log",
+      `Error deleting user ${userName} from ${orgName}: ${err}\n`
+    );
+  }
+}
+
+async function getTeams(orgName) {
+  try {
+    const teams = await octokit.paginate(octokit.teams.list, {
+      org: orgName,
+      per_page: 100,
+    });
+    return teams;
+  } catch (err) {
+    console.log(err);
+    fs.appendFileSync(
+      "cleanup-github-error.log",
+      `Error retrieving teams from ${orgName}: ${err}\n`
+    );
+    return null;
+  }
+}
+
+async function getTeamRepos(orgName, teamName) {
+  try {
+    const repos = await octokit.paginate(octokit.teams.listReposInOrg, {
+      org: orgName,
+      team_slug: teamName,
+      per_page: 100,
+    });
+    return repos;
+  } catch (err) {
+    console.log(err);
+    fs.appendFileSync(
+      "cleanup-github-error.log",
+      `Error retrieving repos from ${orgName} for ${teamName}: ${err}\n`
+    );
+    return null;
+  }
+}
+
+async function getUsersInOrg(orgName) {
+  try {
+    const users = await octokit.paginate(octokit.orgs.listMembers, {
+      org: orgName,
+      per_page: 100,
+    });
+    return users.map((user) => user.login);
+  } catch (err) {
+    console.log(err);
+    fs.appendFileSync(
+      "cleanup-github-error.log",
+      `Error retrieving users from ${orgName}: ${err}\n`
+    );
+    return null;
+  }
+}
+
+async function getAllUsersInTeams(orgName) {
+  const teams = await getTeams(orgName);
+  const users = [];
+  for (let team of teams) {
+    try {
+      const response = await octokit.paginate(octokit.teams.listMembersInOrg, {
+        org: orgName,
+        team_slug: team.name,
+        per_page: 100,
+      });
+      const teamUsers = response.map((user) => user.login);
+      users.push(...teamUsers);
+    } catch (err) {
+      console.log(err);
+      fs.appendFileSync(
+        "cleanup-github-error.log",
+        `Error retrieving users from ${orgName} for ${team.name}: ${err}\n`
+      );
+    }
+  }
+  return users;
+}
+
+async function getTeamsWithNoRepos(orgName, dryRun = true) {
+  const teams = await getTeams(orgName);
+  for (let team of teams) {
+    const repos = await getTeamRepos(orgName, team.name);
+    if (repos !== null && repos.length === 0) {
+      if (!dryRun) {
+        await deleteTeam(orgName, team.name);
+      }
+
+      fs.appendFileSync(
+        "cleanup-github.log",
+        `DRY RUN: Deleting team ${team.name} from ${orgName}\n`
+      );
+    }
+  }
+}
+
+async function getUsersWithNoTeams(orgName, dryRun = true) {
+  const usersInOrg = await getUsersInOrg(orgName);
+  const usersInTeams = await getAllUsersInTeams(orgName);
+  const usersNotInTeams = usersInOrg.filter(
+    (user) => !usersInTeams.includes(user)
+  );
+
+  for (let user of usersNotInTeams) {
+    if (!dryRun) {
+      await deleteUserFromOrg(orgName, user);
+    }
+
+    fs.appendFileSync(
+      "cleanup-github.log",
+      `DRY RUN: Deleting user ${user} from ${orgName}\n`
```

Comment:
nit: conditionally add dry run so we know whether things got really deleted or not

Reply:
Yup, that's controlled by `--dry-run`.

Reply:
no as in add it in the logs as well? so we know clearly which logs refer to dry run which dont
also quick win, lets add time stamps!

Reply:
just wondering out loud, is it trivial to log the ppl in the teams as well? 
hunch here is that i want to log as verbosely as possible since this actually cleans up quite a number of things from our repo.